### ISSUE TITLE
[REQUEST] Hide listview until user has typed

### ISSUE THREAD

**Issue by rcarriga** [davatorium/rofi#1079]
On initial startup the listview can be a bit noisy. It would be great if there was an option to only show elements once the user has typed something (i.e. Filter out all elements if input is empty).

An option like this would work
```css
#listview {
    require-input: true;
}
```
but that is just off the top of my head.

- Rofi version: 1.5.4

**Comment by vincens2005** [comment]
I would also like this

**Comment by DaveDavenport** [comment]
Is the request to hide elements in the listview?  or to hide the listview widget?
I've seen screenshot pasted that suggest the latter, but the issue specifies the first.

**Comment by flame-0** [comment]
Hide the `listview` first and then if you type `x`, it will show the `listview`.

**Example text:**
Initial UI: entry only
```
┌──────────────────────────────────────────────────────────────┐
│ ┌──────────────────────────────────────────────────────────┐ │
│ │ entry: none                                              │ │
│ └──────────────────────────────────────────────────────────┘ │
└──────────────────────────────────────────────────────────────┘
```
Following Initial UI: entry and the results
```
┌──────────────────────────────────────────────────────────────┐
│ ┌──────────────────────────────────────────────────────────┐ │
│ │ entry: hello, world!                                     │ │
│ └──────────────────────────────────────────────────────────┘ │
│ ┌──────────────────────────────────────────────────────────┐ │
│ │ element─text (from: listview)                            │ │
│ └──────────────────────────────────────────────────────────┘ │
│ ┌──────────────────────────────────────────────────────────┐ │
│ │ element─text (from: listview)                            │ │
│ └──────────────────────────────────────────────────────────┘ │
│ ┌──────────────────────────────────────────────────────────┐ │
│ │ element─text (from: listview)                            │ │
│ └──────────────────────────────────────────────────────────┘ │
│ ┌──────────────────────────────────────────────────────────┐ │
│ │ element─text (from: listview)                            │ │
│ └──────────────────────────────────────────────────────────┘ │
└──────────────────────────────────────────────────────────────┘
```

**Example video:**

https://user-images.githubusercontent.com/40175769/164203622-1a742ba7-bc63-4452-a921-0c424b6eef7f.mp4

_example video from https://dribbble.com/shots/14595257-Search-results_

**Comment by DaveDavenport** [comment]
That drawing is not correct, please see `man rofi-theme`. That is the exact reason why I asked the question.
Is the request to hide the whole listview (as in screenshots) or hide the elements (as in request) (this will have significant impact on theming/sizing of window).

from manpage:

```
              ┌────────────────────────────────────────────────────────────────────────────────────┐
              │ window {BOX:vertical}                                                              │
              │ ┌───────────────────────────────────────────────────────────────────────────────┐  │
              │ │ mainbox  {BOX:vertical}                                                       │  │
              │ │ ┌───────────────────────────────────────────────────────────────────────────┐ │  │
              │ │ │ inputbar {BOX:horizontal}                                                 │ │  │
              │ │ │ ┌─────────┐ ┌─┐ ┌───────────────────────────────┐ ┌───┐ ┌───┐ ┌───┐ ┌───┐ │ │  │
              │ │ │ │ prompt  │ │:│ │ entry                         │ │#fr│ │ / │ │#ns│ │ci │ │ │  │
              │ │ │ └─────────┘ └─┘ └───────────────────────────────┘ └───┘ └───┘ └───┘ └───┘ │ │  │
              │ │ └───────────────────────────────────────────────────────────────────────────┘ │  │
              │ │                                                                               │  │
              │ │ ┌───────────────────────────────────────────────────────────────────────────┐ │  │
              │ │ │ message                                                                   │ │  │
              │ │ │ ┌───────────────────────────────────────────────────────────────────────┐ │ │  │
              │ │ │ │ textbox                                                               │ │ │  │
              │ │ │ └───────────────────────────────────────────────────────────────────────┘ │ │  │
              │ │ └───────────────────────────────────────────────────────────────────────────┘ │  │
              │ │                                                                               │  │
              │ │ ┌───────────────────────────────────────────────────────────────────────────┐ │  │
              │ │ │ listview                                                                  │ │  │
              │ │ │ ┌─────────────────────────────────────────────────────────────────────┐   │ │  │
              │ │ │ │ element                                                             │   │ │  │
              │ │ │ │ ┌─────────────────┐ ┌─────────────────────────────────────────────┐ │   │ │  │
              │ │ │ │ │element─icon     │ │element─text                                 │ │   │ │  │
              │ │ │ │ └─────────────────┘ └─────────────────────────────────────────────┘ │   │ │  │
              │ │ │ └─────────────────────────────────────────────────────────────────────┘   │ │  │
              │ │ └───────────────────────────────────────────────────────────────────────────┘ │  │
              │ │                                                                               │  │
              │ │ ┌───────────────────────────────────────────────────────────────────────────┐ │  │
              │ │ │  mode─switcher {BOX:horizontal}                                           │ │  │
              │ │ │ ┌───────────────┐   ┌───────────────┐  ┌──────────────┐ ┌───────────────┐ │ │  │
              │ │ │ │ Button        │   │ Button        │  │ Button       │ │ Button        │ │ │  │
              │ │ │ └───────────────┘   └───────────────┘  └──────────────┘ └───────────────┘ │ │  │
              │ │ └───────────────────────────────────────────────────────────────────────────┘ │  │
              │ └───────────────────────────────────────────────────────────────────────────────┘  │
              └────────────────────────────────────────────────────────────────────────────────────┘
```

**Comment by DaveDavenport** [comment]
vote :+1:  for hide listview and :eyes:  for elements.

**Comment by flame-0** [comment]
> That drawing is not correct, please see `man rofi-theme`

My bad, I just made it simpler.

---

Following your question, I didn't look at the drawing in the documentation properly, so it confused me that `listview` is an independent result. I just realized the `element` is the right independent result inside the `listview`. The right choice here would be to hide the `element` instead of the whole `listview`.
